crates/router/src/core/payments/flows/authorize_flow.rs

```diff
@@ -61,12 +61,12 @@ impl Feature<api::Authorize, types::PaymentsAuthorizeData> for types::PaymentsAu
         mut self,
         state: &AppState,
         connector: &api::ConnectorData,
-        maybe_customer: &Option<domain::Customer>,
```

Comment:
we can remove this field, This is a domain data ( the data that hyperswitch manages ). All the data that the connector needs should be in `RouterData`. We should not be using this customer data anywhere inside this function.

Reply:
These params namely `maybe_customer` & `merchant_account` are being used in `setup_mandate_flow` to fulfill the params of `update_mandate_flow` function.

```
Box::pin(self.update_mandate_flow(
                state,
                merchant_account,
                mandate_id,
                connector,
                key_store,
                call_connector_action,
                &state.conf.mandates.update_mandate_supported,
                connector_request,
                maybe_customer,
                profile_id,
            ))
            .await
```